src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -142,14 +150,14 @@ class DirectStepI : public StepImp<IOUAmount, IOUAmount, DirectStepI>
     void setCacheLimiting (
         IOUAmount const& fwdIn,
         IOUAmount const& fwdSrcToDst,
-        IOUAmount const& fwdOut);
+        IOUAmount const& fwdOut,
+        bool srcRedeems);
 
     friend bool operator==(DirectStepI const& lhs, DirectStepI const& rhs)
     {
         return lhs.src_ == rhs.src_ &&
             lhs.dst_ == rhs.dst_ &&
-            lhs.currency_ == rhs.currency_ &&
-            lhs.noTransferFee_ == rhs.noTransferFee_;
+            lhs.currency_ == rhs.currency_;
```

Comment:
Curious why you are omitting `prevStep_` from the comparison.  Might be good to capture the answer to that question in a comment.